csunplugged/utils/TranslatableModelLoader.py

```diff
@@ -0,0 +1,163 @@
+"""Module for TranslatableModelLoader abstract base class."""
+
+from utils.BaseLoader import BaseLoader
+from utils.language_utils import get_available_languages, get_default_language
+from utils.errors.CouldNotFindConfigFileError import CouldNotFindConfigFileError
+from utils.errors.CouldNotFindMarkdownFileError import CouldNotFindMarkdownFileError
+from utils.errors.InvalidConfigValueError import InvalidConfigValueError
+from utils.errors.MissingRequiredModelsError import MissingRequiredModelsError
+from utils.errors.MissingRequiredFieldError import MissingRequiredFieldError
+from django.utils import translation
+from modeltranslation.utils import fallbacks
+
+
+class TranslatableModelLoader(BaseLoader):
+    """Abstract base class for loaders of translatable models."""
+
+    def get_yaml_translations(self, filename, field_map=None, required_slugs=[], required_fields=[]):
+        """Get a dictionary of translations for the given filename.
+
+        Yaml files must be structured as follows:
+            <object-slug>:
+                <field>: <translated value for field 2>
+                <field2>: <translated value for field 2>
+            <object-slug-2>
+                ...
+
+        Args:
+            filename: (str) path to yaml file from the working directory of the loader
+            field_map: (dict) optional mapping of field names in yaml file to
+                field names in the resulting dictionary
+            required_slugs: (list) list of slugs required to be present in the
+                english strings file
+            required_fields: (list) list of string fields required to be present
+                for each model in the english strings file
+
+        Returns:
+            Dictonary of translations, structured as follows:
+                {
+                    <model-slug>: {
+                        <language>: {
+                            <field-name>:<value>
+                        }
+                    }
+                }
+
+        Raises:
+            CouldNotFindConfigFileError: the requested file could not be found in
+                the /en directory, raised only if required_slugs is not empty.
```

Comment:
Should this be `could not be found in the /structure directory`? I am slightly confused.

Reply:
The `/en` is correct - the error is related to missing yaml files storing the english strings. But you're right to be confused - the exception name is misleading as it now applies to all yaml files, not just config files. I've updated the exception class to reflect this.